hadoop-ozone/dist/src/main/k8s/examples/ozone-csi/datanode-daemonset.yaml

```diff
@@ -0,0 +1,56 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: datanode
+  labels:
+    app.kubernetes.io/component: ozone
+spec:
+  selector:
+    matchLabels:
+      app: ozone
+      component: datanode
+  template:
+    metadata:
+      annotations:
+        prometheus.io/scrape: "true"
+        prometheus.io/port: "9882"
+        prometheus.io/path: /prom
+      labels:
+        app: ozone
+        component: datanode
+    spec:
+      containers:
+      - name: datanode
+        image: '@docker.image@'
+        args:
+        - ozone
+        - datanode
+        ports:
+        - containerPort: 9870
+          name: rpc
+        envFrom:
+        - configMapRef:
+            name: config
+        volumeMounts:
+        - name: data
+          mountPath: /data
+      initContainers: []
+      volumes:
+      - name: data
+        emptyDir: {}
```

Comment:
Can we support mount hostVolume for datanode daemons?

Reply:
Yes, we can.

AFAIK there are two options:
 * using [hostPath](https://kubernetes.io/docs/concepts/storage/volumes/#hostpath)
 * or with [Local PersistentVolumes](https://kubernetes.io/blog/2018/04/13/local-persistent-volumes-beta/)

The first one requires the knowledge of directory names on the host.
The second one is recommended but it requires the creation of PersistentVolumes or install a PersistentVolume provider

I am not sure what is the best approach, my current proposal is:

 * Use empty dir everywhere to make it easier to start simple ozone cluster
 * Provide simple option to turn on any of theses persistence (the kubernetes files are generated and the generation can be parametrized)
 * Document how to customize the kubernetes resources files

Summary: it's question of the defaults:
 
  1. Use a complex, but persistent solution, which may not work out of the box   
  2. Use a simple, but ephemeral solution (as default)

I started to use (2) but I am open to change.

Reply:
Created https://issues.apache.org/jira/browse/HDDS-1646

And we can also discuss it on the community meeting.

Reply:
HostPath would be the right approach, and point the location to a maven build directory.  This will allow maven to clean up the state when data is no longer needed.  This also prevent accidental deletion of user data in default setting.

Reply:
> And point the location to a maven build directory

Unfortunately kubernetes works in a different way, it's not possible:

 * There are multiple nodes, you can't just mount one build directory to multiple nodes
 * The build directory may or may not be on one of the nodes

Reply:
The suggestion of maven build directory in the context of spawning single node k8s cluster in maven for integration test purpose.  In real distributed cluster, mount path is user configurable, and admin is still responsible for keeping the mount path in correct state.

Reply:
Thanks to explain it @eyanghwx I don't think it's possible easily, but feel free to suggest a technical solution at HDDS-1646